### PR TITLE
Fixup deprecated code 3 code path `OpenMP::impl_thread_pool_size`

### PR DESCRIPTION
**Pull request by dalg24**
Fixup for #5836
See https://github.com/kokkos/kokkos/pull/5836/files#r1094841991

cc @stanmoore1 
Reported on Slack

### CODE REVIEW

**Review by masterleinad — COMMENTED** [review]
Shouldn't we pass `*this` to `OpenMP::in_parallel`?

**Review by Rombur — APPROVED** [review]

File: core/src/OpenMP/Kokkos_OpenMP.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ bool OpenMP::in_parallel(OpenMP const &exec_space) noexcept {
 
 int OpenMP::impl_thread_pool_size() const noexcept {
 #ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
-  return OpenMP::in_parallel(exec_space)
+  return OpenMP::in_parallel()
```
Comment (crtrott) [inline core/src/OpenMP/Kokkos_OpenMP.cpp:103]:
```suggestion
  return in_parallel(*this)
```

**Review by crtrott — COMMENTED** [review]
Yeah i think Daniels suggestions sounds good.

**Comment by stanmoore1** [comment]
@dalg24 can you fix the Makefile issue too (shouldn't enable deprecated 3)

**Comment by dalg24** [comment]
> @dalg24 can you fix the Makefile issue too (shouldn't enable deprecated 3)

Not as part of this PR.

Also it is kinda awkward as we do not differentiate between deprecated code 3 and 4
https://github.com/kokkos/kokkos/blob/cba11a18fd70e8cd1c5069c30a9eaad96bab910e/Makefile.kokkos#L523-L526

**Review by Rombur — APPROVED** [review]

File: core/src/OpenMP/Kokkos_OpenMP.cpp (hunk 1 above)
Comment (dalg24) [reply core/src/OpenMP/Kokkos_OpenMP.cpp]: Done

**Comment by stanmoore1** [comment]
> Also it is kinda awkward as we do not differentiate between deprecated code 3 and 4
> 
> https://github.com/kokkos/kokkos/blob/cba11a18fd70e8cd1c5069c30a9eaad96bab910e/Makefile.kokkos#L523-L526

Yeah that is what I mean.

**Review by masterleinad — APPROVED** [review]

**Comment by dalg24** [comment]
Makefile updated in #5850

**Pull request merged**